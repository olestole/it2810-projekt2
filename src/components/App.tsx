import React, { useContext, useEffect } from 'react';
import AppContext from 'utils/AppContext';
import Heading from './Heading/Header';

import ProfileView from '../pages/ProfileView';
import GalleryView from '../pages/GalleryView';

import './main.css';

const App = () => {
  const { appState, appDispatch } = useContext(AppContext);

  useEffect(() => {
    console.log('Satt localStorage');
    localStorage.setItem('context', JSON.stringify(appState));
    document.body.style.backgroundColor = appState.darkmode ? '#121212' : 'white';
  }, [appState]);

  return (
    <div className={appState.darkmode ? 'dark' : 'light'}>
      <Heading />
      {appState.currentUser !== null ? <ProfileView /> : <GalleryView />}
    </div>
  );
};

export default App;
