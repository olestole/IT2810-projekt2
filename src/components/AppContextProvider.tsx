import React, { useReducer } from 'react';
import { initialAppState } from 'utils/AppContext';
import AppContext from 'utils/AppContext';
import reducer from 'utils/AppReducer';

const AppContextProvider = ({ children }: any) => {
  let localStorageState;

  if (localStorage.getItem('context')) {
    localStorageState = JSON.parse(localStorage.getItem('context') as string);
  }
  const [appState, appDispatch] = useReducer(reducer, localStorageState ? localStorageState : initialAppState);

  return <AppContext.Provider value={{ appState, appDispatch }}>{children}</AppContext.Provider>;
};

export default AppContextProvider;
