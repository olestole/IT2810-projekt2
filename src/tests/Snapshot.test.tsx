import React from 'react';
import '@testing-library/jest-dom/extend-expect';

import * as renderer from 'react-test-renderer';
import AppContextProvider from '../components/AppContextProvider';
import App from '../components/App';

describe('Whole page', () => {
  it('matches the snapshot', () => {
    const wholePage = renderer.create(
      <AppContextProvider>
        <App />
      </AppContextProvider>,
    );

    expect(wholePage).toMatchSnapshot();
  });
});
