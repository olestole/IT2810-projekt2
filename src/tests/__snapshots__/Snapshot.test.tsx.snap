// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whole page matches the snapshot 1`] = `
<div
  className="light"
>
  <div>
    <header
      className="header"
    >
      <div
        className="logo"
      >
        <img
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": "IconMock",
            }
          }
        />
      </div>
      <div
        className="darkmode"
      >
        <img
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": "IconMock",
            }
          }
        />
      </div>
    </header>
  </div>
  <div>
    <div
      className="filterBoxCollection"
    >
      <div
        className="filterBox"
      >
        <p
          className="checkBoxText"
        >
          cloud
        </p>
        <input
          checked={false}
          className="checkBox"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="filterBox"
      >
        <p
          className="checkBoxText"
        >
          stars
        </p>
        <input
          checked={false}
          className="checkBox"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="filterBox"
      >
        <p
          className="checkBoxText"
        >
          spaceship
        </p>
        <input
          checked={false}
          className="checkBox"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="filterBox"
      >
        <p
          className="checkBoxText"
        >
          sun
        </p>
        <input
          checked={false}
          className="checkBox"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="filterBox"
      >
        <p
          className="checkBoxText"
        >
          heart
        </p>
        <input
          checked={false}
          className="checkBox"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="filterBox"
      >
        <p
          className="checkBoxText"
        >
          male
        </p>
        <input
          checked={false}
          className="checkBox"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="filterBox"
      >
        <p
          className="checkBoxText"
        >
          female
        </p>
        <input
          checked={false}
          className="checkBox"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
    </div>
    <div
      className="gridContainer"
    >
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="stars"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Toby
        </p>
      </div>
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="spaceship"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Astrid
        </p>
      </div>
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="cloud"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Thomas
        </p>
      </div>
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="sun"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Selina
        </p>
      </div>
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="stars"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Albert
        </p>
      </div>
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="sun"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Mary
        </p>
      </div>
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="cloud"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Bill
        </p>
      </div>
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="spaceship"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Sophie
        </p>
      </div>
      <div
        className="svgContainer"
        onClick={[Function]}
      >
        <div
          className="svgAnimation"
          id="cloud"
        >
          <IconMock />
        </div>
        <div
          className="svgMainElements"
        >
          <div
            id="heart"
          />
          <div
            id="people"
          >
            <IconMock />
          </div>
        </div>
        <p
          className="svgText"
        >
          Harry
        </p>
      </div>
    </div>
  </div>
</div>
`;
